addon/-private/system/relationships/state/has-many.js

```diff
@@ -209,7 +209,7 @@ export default class ManyRelationship extends Relationship {
 
     for (let i = 0; i< internalModels.length; i++) {
       let internalModel = internalModels[i];
-      if (this.canonicalMembers.has(internalModel)) {
```

Comment:
I'm super confused how these end up being different. So  thank you for providing a test which I can debug :)